src/blockdata/transaction.rs

```diff
@@ -608,7 +608,23 @@ impl Decodable for Transaction {
     }
 }
 
-/// Hashtype of a transaction, encoded in the last byte of a signature
+/// Error occuring when parsing a SigHashType
+#[derive(Debug, PartialEq)]
+pub enum ParseSigHashTypeError {
+    /// This type is consensus valid but an input including it would prevent the transaction from
+    /// being relayed on today's Bitcoin network.
+    NonStandard,
```

Comment:
Will there ever be a second error type? If not maybe a zero sized struct might be an option too? Extending this enum is an API break anyway, so switching to an enum wouldn't be a problem in that (unlikely) case.

Reply:
Good point, i wanted to make this somewhat extensible but it would break the API anyways (and a new variant is very unlikely).